packages/core/src/tracing/instrumentation/protocols/nativeFetch.js

```diff
@@ -96,15 +95,19 @@ function instrument() {
       originalArgs[i] = arguments[i];
     }
 
-    if (skipTracingResult.skip) {
+    if (
+      skipTracingResult.skip ||
+      (!skipTracingResult.allowRootExitSpan && !parentSpan) ||
+      (!skipTracingResult.allowRootExitSpan && skipTracingResult.isExitSpan)
+    ) {
```

Comment:
I can't follow this change. Too much in one commit without any explanation.

Reply:
Discussed: Revert and create separate commit with proper explanation

Reply:
Added a comment for the current change